src/lib/storage/index/partial_hash/partial_hash_index.cpp

```diff
@@ -0,0 +1,80 @@
+#include "partial_hash_index.hpp"
+
+#include "storage/segment_iterate.hpp"
+
+namespace hyrise {
+
+PartialHashIndex::PartialHashIndex(const std::vector<std::pair<ChunkID, std::shared_ptr<Chunk>>>& chunks_to_index,
+                                   const ColumnID column_id)
+    : _column_id{column_id} {
+  Assert(!chunks_to_index.empty(), "PartialHashIndex requires chunks_to_index not to be empty.");
+  resolve_data_type(chunks_to_index.front().second->get_segment(column_id)->data_type(),
+                    [&](const auto column_data_type) {
+                      using ColumnDataType = typename decltype(column_data_type)::type;
+                      _impl = std::make_unique<PartialHashIndexImpl<ColumnDataType>>(chunks_to_index, _column_id);
+                    });
+}
+
+size_t PartialHashIndex::insert_entries(
```

Comment:
Makes sense.

Reply:
We don't insert chunks, we create index entries for the chunks passed as parameters. `insert/delete_chunks` would be misleading IMHO. `insert_entries_for_chunks` and `remove_entries_of_chunks` would be more descriptive but less concise. With the parameter types and names, one could argue that `for_chunks` and `of_chunks` is redundant.

Reply:
Hmm, I don't agree. Partially because "entries" is not really a useful term here that imo. Maybe `add_chunk_to_index` (or simply `add()` which only takes chunks?) would be better? I don't think `insert_entries` makes clear what those entries are and what is considered an "entry".

It's absolutely no show stopper. But I stumpled over these "entries" (do they refer to values? To positions? To keys?) which are apparently never used outside `add` and `remove` functions.